-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b114a4/24393/summary.html
**COMMIT**: 78ad53115be9049aa1ac3b809e129de3fed1b753
**CMSSW**: CMSSW_12_4_X_2022-05-02-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/37745/24393/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>@@@@ Running edmWriteConfigs for CalibCalorimetryCaloTPGPlugins
--- Registered EDM Plugin: CalibCalorimetryCaloTPGPlugins
Entering library rule at CalibCalorimetry/CaloTPG
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_4_X_2022-05-02-1100/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_4_X_2022-05-02-1100/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc: In member function 'void CaloTPGTranscoderULUT::loadHCALCompress(const HcalLutMetadata&, const HcalTrigTowerGeometry&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_4_X_2022-05-02-1100/src/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc:88:36: error: conversion from 'unsigned int' to '__gnu_cxx::__alloc_traits<std::allocator<unsigned char>, unsigned char>::value_type' {aka 'unsigned char'} changes value from '256' to '0' [-Werror=overflow]
   88 |             outputLUT_[index][i] = TPGMAX;
      |                                    ^~~~~~
cc1plus: some warnings being treated as errors
gmake: *** [tmp/slc7_amd64_gcc10/src/CalibCalorimetry/CaloTPG/src/CalibCalorimetryCaloTPG/CaloTPGTranscoderULUT.cc.o] Error 1
>> Building  shared library tmp/slc7_amd64_gcc10/src/CalibCalorimetry/CaloTPG/src/CalibCalorimetryCaloTPG/libCalibCalorimetryCaloTPG.so
</pre>

## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

--- a/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
+++ b/CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc
@@ -83,6 +83,10 @@ void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
       for (unsigned int i = threshold; i < lutsize; ++i)
         if (allLinear_) {
           outputLUT_[index][i] = isOnlyQIE11(id) ? linearQIE11LUT[i] : linearQIE8LUT[i];
+          //Modifying the saturation (127 -> 255) for the 'split cells'.
+          if (abs(ieta) > 20 && isOnlyQIE11(id) && linearQIE11LUT[i] >= (TPGMAX - 2) / 2.) {
+            outputLUT_[index][i] = TPGMAX - 1;
+          }
         } else {
           outputLUT_[index][i] = analyticalLUT[i];
         }
@@ -104,6 +108,10 @@ void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
           if (outputLUT_[index][i] != tpg) {
             tpg = outputLUT_[index][i];
             hcaluncomp_[index][tpg] = lsb_factor_ * i / (isOnlyQIE11(id) ? lin11_factor_ : lin8_factor_);
+            //Modifying the saturation for the 'split cells'
+            if (abs(ieta) > 20 && isOnlyQIE11(id) && linearQIE11LUT[i] >= (TPGMAX - 2) / 2.) {
+              hcaluncomp_[index][tpg] = (TPGMAX - 1) / 2.;
+            }
           }
         }
       } else {

